torch/_subclasses/fake_tensor.py

```diff
@@ -509,6 +530,168 @@ def convert(t, mem_fmt):
             )
 
 
+FAST_OP_IMPLEMENTATIONS = {}
+
+
+# Unlike register_op_impl, these don't do the slow iteration for
+# run_impl_check, and these run BEFORE decompositions
+def register_fast_op_impl(func: OpOverload):
+    def impl_decorator(op_impl):
+        FAST_OP_IMPLEMENTATIONS[func] = op_impl
+        return op_impl
+
+    return impl_decorator
+
+
+# infer_size_impl in ExpandUtils
+def infer_size(a, b):
+    dimsA = len(a)
+    dimsB = len(b)
+    ndim = max(dimsA, dimsB)
+    expandedSizes = [0] * ndim
+    for i in range(ndim - 1, -1, -1):
+        offset = ndim - 1 - i
+        dimA = dimsA - 1 - offset
+        dimB = dimsB - 1 - offset
+        sizeA = a[dimA] if dimA >= 0 else 1
+        sizeB = b[dimB] if dimB >= 0 else 1
+        assert sizeA == sizeB or sizeA == 1 or sizeB == 1
+        expandedSizes[i] = sizeB if sizeA == 1 else sizeA
+    return tuple(expandedSizes)
+
+
+def make_fast_binary_impl(slow_ref):
+    def fast_binary_impl(mode, *args, **kwargs):
+        def slow(msg):
+            count_label(f"slow {msg}")
+            with mode:
+                return slow_ref(*args, **kwargs)
+
+        count_label("attempt fast")
+        # == Fast path (based off of TensorIterator fast path) ==
+        operands = args
+
+        # compute_shape
+        has_scalars = False
```

Comment:
it's a lot of code duplication of logic that exists elsewhere :/ but if it's much slower to reuse code, i guess that's life. In any case, could we test that ?

Reply:
I'm working on the testing right now. It's slower to reuse code: PrimTorch doesn't actually implement a fastpath, and this is different logic than TensorIterator.

Reply:
Where is the C++ -> python LLM when you need it 🤡 🤪